server/game/cards/12-SoW/AnOceanInADrop.js

```diff
@@ -0,0 +1,51 @@
+const DrawCard = require('../../drawcard.js');
+const { Locations, TargetModes } = require('../../Constants');
+const AbilityDsl = require('../../abilitydsl.js');
+
+class AnOceanInADrop extends DrawCard {
+    setupCardAbilities() {
+        this.action({
+            title: 'Place hand on bottom of deck and draw cards',
+            condition: context => this.game.isDuringConflict() && context.source.parent && context.source.parent.isParticipating(),
+            cost: AbilityDsl.costs.sacrificeSelf(),
+            target: {
+                mode: TargetModes.Select,
+                targets: true,
+                choices:  {
+                    [this.owner.name]: AbilityDsl.actions.sequential(this.getGameActions(this.owner)),
+                    [this.owner.opponent && this.owner.opponent.name || 'NA']: AbilityDsl.actions.sequential(this.getGameActions(this.owner.opponent))
+                }
+            },
+            effect: 'place {1}\'s hand on the bottom of their deck and have {1} draw {2} cards',
+            effectArgs: (context) => context.select === this.owner.name ?
+                [this.owner.name, context.player.hand.value().length] :
+                [this.owner.opponent.name, context.player.opponent.hand.value().length]
+        });
+    }
+
+    getGameActions(player) {
```

Comment:
you're passing `player` but you never use it - shouldn't all your references to `target` actually be to `player`?